### PR TITLE
feature(alternator): adding a 48h alternator test

### PR DESCRIPTION
**Pull request by fruch**
now that was have the 3h test case up and running with nemesis,
we want have weekly runs with the 48h case.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [ ] I didn't leave commented-out/debugging code
- [ ] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
**NOTICE** this depend on #1748 to actually work correctly

**Comment by fgelcer** [comment]
@roydahan , ping

**Comment by bentsi** [comment]
@fgelcer please check the CI before approving PRs:

image

File: defaults/test_default.yaml [hunk 1]
```diff
@@ -56,3 +56,7 @@ email_recipients: ['qa@scylladb.com']
 collect_logs: false
 
 loader_swap_size: 1024 # Size of swap file for loader node in bytes 1024 * 1MB
+
+alternator_enforce_authorization: false
+alternator_access_key_id: ''
```
Comment (roydahan) [inline defaults/test_default.yaml]: defaults are empty?

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+test_duration: 180
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: why 180?
it's not enough for 48h

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+prepare_write_cmd: [
+  "bin/ycsb load dynamodb -P workloads/workloada -threads 250 -p recordcount=200200300 -p fieldcount=1 -p fieldlength=1024 -p dataintegrity=true -p hdrhistogram.summary.addstatus=true", ]
+stress_cmd: [
+  "bin/ycsb run  dynamodb -P workloads/workloada -threads 250 -p recordcount=200200300 -p fieldcount=1 -p fieldlength=1024 -p operationcount=400400600 -p dataintegrity=true -p hdrhistogram.summary.addstatus=true", ]
+
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: What is the load in the cluster in terms of CPU when this stress run?

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+dynamodb_primarykey_type: HASH_AND_RANGE
+
+n_loaders: 3
+instance_type_db: 'i3.4xlarge'
+n_monitor_nodes: 1
+n_db_nodes: 4
+
+nemesis_class_name: 'LimitedChaosMonkey'
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: Change to ChaosMonkey.
Limited is only when using CL=ONE.

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+nemesis_interval: 30
+nemesis_during_prepare: false
+
+user_prefix: 'alternator-3h'
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: fix user prefix to 48h

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+test_duration: 180
+prepare_write_cmd: [
+  "bin/ycsb load dynamodb -P workloads/workloada -threads 250 -p recordcount=200200300 -p fieldcount=1 -p fieldlength=1024 -p dataintegrity=true -p hdrhistogram.summary.addstatus=true", ]
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: I understand you copied it from the current c-s but I don't know if it make sense.
You need to consult the team about a typical schema of dynamo or google it.
I don't think that fieldcount=1 and fieldlength=1024 is something that will help us to catch real life bugs.

Comment (roydahan) [reply test-cases/longevity/longevity-alternator-200GB-48h.yaml]: Also, why do we need a prepare?
Is this stage a write only?

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+stress_cmd: [
+  "bin/ycsb run  dynamodb -P workloads/workloada -threads 250 -p recordcount=200200300 -p fieldcount=1 -p fieldlength=1024 -p operationcount=400400600 -p dataintegrity=true -p hdrhistogram.summary.addstatus=true", ]
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: Same as above + i'm not sure we actually need to separate into two stages...

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+nemesis_class_name: 'LimitedChaosMonkey'
+nemesis_interval: 30
+nemesis_during_prepare: false
+
+user_prefix: 'alternator-3h'
+
+alternator_port: 8080
+alternator_enforce_authorization: true
+alternator_access_key_id: 'alternator'
```
Comment (roydahan) [inline test-cases/longevity/longevity-alternator-200GB-48h.yaml]: Why aren't these the defaults and if need to be changed we'll override them...
This is how it's been till now.

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml (hunk 3 above)
Comment (fruch) [reply test-cases/longevity/longevity-alternator-200GB-48h.yaml]: we don't know we didn't yet run this.

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml (hunk 6 above)
Comment (fruch) [reply test-cases/longevity/longevity-alternator-200GB-48h.yaml]: Yes this is write only, as for sizes, we just know dynamodb has 400K limit per request

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml (hunk 7 above)
Comment (fruch) [reply test-cases/longevity/longevity-alternator-200GB-48h.yaml]: for using dataintegrity=true, we have to

File: test-cases/longevity/longevity-alternator-200GB-48h.yaml (hunk 8 above)
Comment (fruch) [reply test-cases/longevity/longevity-alternator-200GB-48h.yaml]: the default are not to use authorization, `alternator_enforce_authorization` is in scylla yaml
`alternator_access_key_id` is what we pass into the client.
it's following the same logic as `client_encrypt` and `server_encrypt` the defaults are false.

**Comment by fruch** [comment]
@roydahan, I've me and @fgelcer discussed, @fgelcer would take this PR forward, and will re-open it with the needed changes

**Pull request closed without merging**